### PR TITLE
fix(interactive): Support Statistics in Groot Storage

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

As titled.

## Related issue number

#3843

### CODE REVIEW

**Review by siyuan0322 — APPROVED** [review]

File: interactive_engine/groot-server/src/main/java/com/alibaba/graphscope/groot/servers/ir/GrootMetaFetcher.java [hunk 1]
```diff
@@ -42,13 +47,14 @@ public Optional<IrMeta> fetch() {
...
-            try {
-                // TODO: add statistics, otherwise, the CBO will not work
-                return Optional.of(
-                        new IrMeta(
-                                new SnapshotId(true, snapshotId), new IrGraphSchema(schema, true)));
-            } catch (IOException e) {
-                throw new RuntimeException(e);
+            SnapshotId si = new SnapshotId(true, snapshotId);
+            IrGraphSchema irSchema = new IrGraphSchema(schema, true);
+            GraphStatistics statistics = this.schemaFetcher.getStatistics();
```
Comment (longbinlai) [inline interactive_engine/groot-server/src/main/java/com/alibaba/graphscope/groot/servers/ir/GrootMetaFetcher.java:52]: It's not okay to place getting statistics with schema for now. For larger graph, getting statistics is a very time-costly operation, and it is operated at the frequency as schema? I don't think so.

**Review by longbinlai — COMMENTED** [review]
Please separate getting statistics from getting schema.

File: interactive_engine/executor/store/groot/src/db/api/schema.rs [hunk 2]
```diff
@@ -360,6 +364,74 @@ impl PropDef {
...
+        edge_type_count: HashMap<EdgeKind, i64>,
+    ) -> Self {
+        GraphPartitionStatistics { version, vertex_count, edge_count, vertex_type_count, edge_type_count }
+    }
+
+    pub fn get_version(&self) -> i64 {
+        self.version
+    }
+
+    pub fn get_vertex_count(&self) -> i64 {
```
[the remark this answers is not in the document]
Comment (BingqingLyu) [reply interactive_engine/executor/store/groot/src/db/api/schema.rs:388]: It is refined to be `u64`.

File: interactive_engine/groot-server/src/main/java/com/alibaba/graphscope/groot/servers/ir/GrootMetaFetcher.java (hunk 1 above)
Comment (BingqingLyu) [reply interactive_engine/groot-server/src/main/java/com/alibaba/graphscope/groot/servers/ir/GrootMetaFetcher.java]: We have separated the two interfaces for getting schema (+procedures) and statistics.

**Review by siyuan0322 — DISMISSED** [review]

File: interactive_engine/groot-module/src/main/java/com/alibaba/graphscope/groot/coordinator/SchemaManager.java [hunk 3]
```diff
@@ -78,11 +102,97 @@ public void start() {
...
                                 "recover", logger));
-        this.scheduler.scheduleWithFixedDelay(this::recover, 5, 2, TimeUnit.SECONDS);
+        this.syncSchemaScheduler.scheduleWithFixedDelay(this::recover, 5, 2, TimeUnit.SECONDS);
+
+        if (this.collectStatistics) {
+            this.fetchStatisticsScheduler =
+                    Executors.newSingleThreadScheduledExecutor(
+                            ThreadFactoryUtils.daemonThreadFactoryWithLogExceptionHandler(
+                                    "fetch-statistics", logger));
+            this.fetchStatisticsScheduler.scheduleWithFixedDelay(
```
[the remark this answers is not in the document]
Comment (siyuan0322) [reply interactive_engine/groot-module/src/main/java/com/alibaba/graphscope/groot/coordinator/SchemaManager.java:115]: fixe

**Review by longbinlai — APPROVED** [review]
LGTM

**Pull request merged**